### DISCUSSION TITLE
OpenWebUI local machine hardware

### DISCUSSION THREAD

**Discussion by stevenbaert** [open-webui/open-webui#4519] [General]
Hi,

I'm looking for advice on home hardware to run Open-WebUI with Ollama Docker (Llama 3.1, preferably 70B). I'd like to enable Coqui voice, so I assume an Nvidia GPU is necessary. I'm considering a budget NUC or mini PC that's not too noisy, with a budget up to $2,000, though I'd prefer to stay around $1,000 if possible.

Note: I have a Synology 923+ (24GB ram) which runs my Docker images and already an older mini pc with 64GB ram but that one has a an Intel GPU and so Llama 3.1 on that machine is really slow and all cpu is taken.

Any suggestions?

Thanks!  
S

**Comment by rztaylor** [comment]
You might find this reddit post useful: https://www.reddit.com/r/LocalLLaMA/comments/1eiwnqe/hardware_requirements_to_run_llama_3_70b_on_a/#:~:text=The%20official%20Meta%20website%20recommends,a%203000%20series%20graphics%20card.

**Comment by stevenbaert** [comment]
Thanks for the info! I’m hoping to find someone who already has it set up and running smoothly to avoid any unnecessary issues. While 70B sounds ideal, I’m open to using the 8B model if budget constraints are a concern.

So, in short: If I go with a Mac, I won't be able to run Coqui locally, right? But it would be much simpler and straightforward—maybe even a Mac Mini would do the job? On the other hand, if I opt for Nvidia, I’d likely need a Windows server (which I have, but no NVidia card and no way to add it either), which comes with its own challenges like noise, power consumption, and fan management. What do you think?

**Reply by rztaylor** [reply]
There's a lot of options, and it really depends what your use case is, so here are a few random thoughts based on my own adventures. 

I run Open WebUI on a home server small kubernetes cluster of mostly mini PCs that I bought off ebay. Myself and a few friends predominantly use it to give us access to a a great interface/features whilst remaining provider/model agnostic. We use an OpenAI API key, and with our (admittedly fairly low) usage we've spent less than $30 over an 8 months period. Obviously this is far, far cheaper that buying new hardware - but that's text only and you're interested in TTS.

For image generation, I use stable disffusion with ComfyUI on my gaming machine which has a RTX 3070ti and I find it absolutely fine for my needs. Running Ollama with Llama3 on that card is also very responsive. However, it's an expensive route if you don't already have a GFX card. My gaming machine cost $900 (self build with core i5, 64GB RAM, RTX3070ti, 2TB SSD). I can't quote any running costs on this, which obviously you'd need to factor in. If you're into gaming, at least it has dual use  :)

If you don't want a GFX card and will rely on CPU only then the Apple M processors are the best bet (for now at least), but that comes with it's own problems in terms of what software will run. 

Mini PC's, such as https://www.bosgamepc.com/products/mini-pc-g35?type=feature might be something you could use (I do have one of these, which is amazing for it's size) but you'd need to see if you can find benchmarks for the specific Intel i5 it uses to get an idea of how it would perform.

Lastly, with the current pace in AI, and the hardware that is being launched specifically for it, such as the NPU in the Snapdragon X Elite  I feel this is probably a bad time to invest a lot of money in hardware. But this is only my opinion, so if you're happy to spend then don't let me put you off!

**Reply by stevenbaert** [reply]
What about Nvidia Jetson hardware? Any experience?  Seems perfect for the job? 



https://www.amazon.com/NVIDIA-Jetson-Orin-Nano-Developer/dp/B0BZJTQ5YP/ref=mp_s_a_1_1?crid=1E1TB3AKALHJK&dib=eyJ2IjoiMSJ9.WXnsXcibbbRFZBJ_cz2X3B3cf0swRJOW8Jv9hsVUGLNuJf-QQV7srC-JJklqAjbH5iaGx33F34mkGcCcfnR4w37VL-U_ntM2A08xV-jdmFTZCfYj1azomvNdvzG09z7Lp3LdSHu1yaQVt9OdLnkmDJkVJxWB6ScDNZNcxXg7NLogsCFqNhll9nz4__G3oUlw6Ps9lAs965kNgmqqeM78oQ.2KhtpQbNqBxuW-8I15GVqDlOZ8m1TdhgAnHOPrtcc_w&dib_tag=se&keywords=nvidia+jetson&qid=1723574529&refinements=p_36%3A-83000&rnid=386442011&sprefix=Nvidia+jetzon%2Caps%2C201&sr=8-1

Or

https://www.amazon.com.be/-/nl/NVIDIA-Jetson-Orin-Nano-Development/dp/B0BZJTQ5YP/ref=asc_df_B0BZJTQ5YP/?tag=begogshpads0a-21&linkCode=df0&hvadid=655693703953&hvpos=&hvnetw=g&hvrand=5538293457636184089&hvpone=&hvptwo=&hvqmt=&hvdev=m&hvdvcmdl=&hvlocint=&hvlocphy=1001175&hvtargid=pla-2112361227514&psc=1&mcid=2d5410c6b9d13d94ba6822aa723d3a23
